src/config/config_poll.rs

```diff
@@ -0,0 +1,98 @@
+use std::collections::HashMap;
+use std::sync::{Arc, RwLock};
+use std::time::Duration;
+
+use hyper::header::HeaderValue;
+use reqwest::header::IF_NONE_MATCH;
+use reqwest::Client;
+use tokio::time;
+
+use crate::blueprint::Blueprint;
+use crate::cli::CLIError;
+use crate::config::Config;
+use crate::http::ServerContext;
+
+pub struct ConfigLoader {
+  file_path: String,
+  refresh_interval: u64,
+  state: Arc<RwLock<ServerContext>>,
```

Comment:
I think we should use a non-blocking implementation instead of RWLock. RWLock can drastically affect the server's performance, even when this feature is disabled.

Reply:
Moreover we should have two separate implementations, one that just load's once and doesn't ever update. This doesn't require any locking while reads. The other should be implemented using an `AtomicPtr` for better performance.